src/libsyntax/print/pprust.rs

```diff
@@ -180,6 +162,31 @@ fn binop_to_string(op: BinOpToken) -> &'static str {
     }
 }
 
+fn literal_to_string(lit: token::Lit, suffix: Option<ast::Name>) -> String {
+    let mut out = match lit {
+        token::Byte(b)           => format!("b'{}'", b),
+        token::Char(c)           => format!("'{}'", c),
+        token::Err(c)            => format!("'{}'", c),
```

Comment:
Hm, I think we no longer use the `Err` token? Perhaps it makes sense to purge it in this PR as well?

I can imagine that, if we start using `Err` again, this `format!("'{}'", c),` might break (in a sense that it'll produce more than one token)

Reply:
`token::Err` is needed at least as a token counterpart to `LitKind::Err` (similarly to `token::Bool`).

Beside that, I have some plans for it.
It would be good for error recovery to have an "error token" that proc macros can accept and produce (AST based macros use `DummyResult::any`/`DummyResult::expr` for this to great success, see e.g. https://github.com/rust-lang/rust/pull/57205 and the issue it fixes).
A literal that's lexically invalid, but can survive through stringification would probably be a good candidate for such token.
(This needs some thought, so I'm unlikely to touch it in this PR.)